src/Parser/DemonHunter/Vengeance/Modules/Spells/DemonSpikes.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import SCHOOLS from 'common/MAGIC_SCHOOLS';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+class DemonSpikes extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  hitsWithDS = 0;
+  hitsWithoutDS = 0;
+  hitsWithDSOnCD = 0;
+
+  on_toPlayer_damage(event) {
+    // Physical
+    if (event.ability.type !== SCHOOLS.ids.PHYSICAL) {
+      return;
+    }
+    if (this.combatants.selected.hasBuff(SPELLS.DEMON_SPIKES_BUFF.id, event.timestamp)) {
+      this.hitsWithDS += 1;
+    }else{
+      this.hitsWithoutDS += 1;
+
+      const onCooldown = this.spellUsable.isOnCooldown(SPELLS.DEMON_SPIKES.id);
+      if(onCooldown) {
+        this.hitsWithDSOnCD += 1;
+      }
+    }
+  }
+
+  get mitigatedUptime(){
+    return formatPercentage(this.hitsWithDS / (this.hitsWithDS + this.hitsWithoutDS));
+  }
+
+  statistic() {
+    const demonSpikesUptime = this.combatants.selected.getBuffUptime(SPELLS.DEMON_SPIKES_BUFF.id);
+
+    const demonSpikesUptimePercentage = demonSpikesUptime / this.owner.fightDuration;
+
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.DEMON_SPIKES.id} />}
+        value={`${this.mitigatedUptime}%`}
+        label="Hits Mitigated by Demon Spikes"
+        tooltip={`Demon Spikes usage breakdown:
+          <ul>
+          <li>You were hit <b>${this.hitsWithDS}</b> times with your Demon Spikes buff.</li>
+          <li>You were hit <b>${this.hitsWithoutDS}</b> times <b><i>without</i></b> your Demon Spikes buff.</li>
+          <li>You were hit <b>${this.hitsWithDSOnCD}</b> times <b><i>with</i></b> Demon Spikes avalible for use but not used.</li>
+          </ul>
+          <b>${this.mitigatedUptime}%</b> of physical attacks were mitigated with Demon Spikes, and your overall uptime was <b>${formatPercentage(demonSpikesUptimePercentage)}%</b>.`}
```

Comment:
Don't really need to have the mitigatedUptime here since it is already the primary value.

Reply:
Yea your right, took it out.